app/services/search_orders.rb

```diff
@@ -15,11 +15,16 @@ def initialize(params, current_user)
   attr_reader :params, :current_user
 
   def fetch_orders
-    @search = search_query.
+    search = search_query.
       includes(:payments, :subscription, :shipments, :bill_address, :distributor, :order_cycle).
-      ransack(params[:q])
+      ransack(params[:q]).
+      result(distinct: true)
 
-    @search.result(distinct: true).joins(:bill_address)
+    if params.dig(:q, :s)&.starts_with?("bill_address_")
+      search = search.select('spree_addresses.*, spree_orders.*')
+    end
```

Comment:
This parameter comes from the bulk order management via the API controller it's using Ransack features to find the attribute on the association. Ransack seems to add a join for this and we would end up with two joins if we added `references` here as well. So we don't and just need to help Ransack with the select.